uiowa/uiowa
# initial commit of Student Life Migrate

Counterpart to our SilverStripe XML export module: https://github.com/StudentLifeMarketingAndDesign/silverstripe-migration-tools

This module parses pages, posts, people, categories, and files (PDFs, etc) and actively used images that are exported from our SilverStripe sites and imports them into SiteNow.

# How to test

Example SilverStripe site:

Content XML file: [content.xml.zip](https://github.com/uiowa/uiowa/files/7204923/content.xml.zip)
Website the XML file comes from: https://spock.imu.uiowa.edu/student-space/ (for reference)

1.  Download an XML file provided by one of the SilverStripe sites (example attached above)
2. Place the XML file in uiowa/files-private/migrations/current/content.xml
3. Enable studentlife_migrate (and dependencies) in local Drupal environment (Sandbox in this example): `drush @sandbox.local en studentlife_migrate -y`
4. Run `drush @sandbox.local migrate:import studentlife_content_staffpage,studentlife_images_import,studentlife_images_to_media,studentlife_categories,studentlife_tags,studentlife_content_page,studentlife_content_post,studentlife_files_import_remote,studentlife_files_to_media_remote`
5. Ensure that the pages, posts, people, tags, files, and actively used images have been migrated (there should be at least one of each of those in the example:

- Pages:
  - Home
  - Available Spaces
  - Example StaffHolderPage
  - Example Blog
- Posts:
  - Example Post
 - People:
   - Dustin Quam
 - Tags: 
   - Test tag
   - Another tag
 - Files:
   -  New-Attorney-Referral-List.pdf
  - Images: 
    - IMU-General2__ResizedImageWzYwMCwzMzhd.jpg
    - 165149720_10108953978731789_669086232225469991_n__FitMaxWzI1OTIsMTQ1OF0.jpg (Dustin wearing a rain coat)

Things that don't quite work right:
* Images are only imported and not attached to the nodes themselves and need to be manually linked
* Same with files, files are merely imported into the media library and usually have to be manually linked up
* There's always a duplicate of the final post/page imported,possibly due to: https://www.drupal.org/project/migrate_plus/issues/2891964
